server/src/main/java/org/elasticsearch/cluster/DiffableUtils.java

```diff
@@ -104,8 +104,8 @@ public static <K, T, M extends Map<K, T>> MapDiff<K, T, M> emptyDiff() {
      */
     @SuppressWarnings("unchecked")
     public static <K, T extends Diffable<T>, T1 extends T, T2 extends T, M extends Map<K, T>> MapDiff<K, T, M> merge(
-        MapDiff<K, T1, ? extends ImmutableOpenMap<K, T1>> diff1,
-        MapDiff<K, T2, ? extends ImmutableOpenMap<K, T2>> diff2,
+        MapDiff<K, T1, ? extends Map<K, T1>> diff1,
+        MapDiff<K, T2, ? extends Map<K, T2>> diff2,
```

Comment:
Are changes in this file necessary for this PR? If not, I'd prefer it remains unchanged and you can raise a separate PR for pure refactor. PS, I also wonder whether the choice of `ImmutableOpenMap` is intentional in this class since we tend to ImmutableOpenMap for diff building.